components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaFullscreenOrientationFeature.kt

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.media.fullscreen
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.pm.ActivityInfo
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import mozilla.components.browser.state.state.MediaState.FullscreenOrientation
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * Feature that will auto-rotate the device to the correct orientation for the media aspect ratio.
+ */
+class MediaFullscreenOrientationFeature(
```

Comment:
This looks nice! I think eventually we could merge this with `FullScreenFeature`. But since that is not migrated to `BrowserStore` yet ... not in this PR. :)